neurodamus/utils/memory.py

```diff
@@ -160,6 +161,159 @@ def pretty_printing_memory_mb(memory_mb):
         return "%.2lf PB" % (memory_mb / 1024 ** 3)
 
 
+@run_only_rank0
+def distribute_cells(dry_run_stats, num_ranks, batch_size=10) -> (dict, dict):
+    """
+    Distributes cells across ranks based on their memory load.
+
+    This function uses a greedy algorithm to distribute cells across ranks such that
+    the total memory load is balanced. Cells with higher memory load are distributed first.
+
+    Args:
+        dry_run_stats (DryRunStats): A DryRunStats object.
+        num_ranks (int): The number of ranks.
+
+    Returns:
+        rank_allocation (dict): A dictionary where keys are rank IDs and
+                                values are lists of cell IDs assigned to each rank.
+        rank_memory (dict): A dictionary where keys are rank IDs
+                            and values are the total memory load on each rank.
+    """
+    logging.debug("Distributing cells across %d ranks", num_ranks)
+
+    def validate_inputs(dry_run_stats, num_ranks, batch_size):
```

Comment:
Could this be a method of `DryRunStats`, i.e., we could just call `dry_run_stats.validate()`? Raise a `ValueError` if things go south?

Reply:
The `validate_inputs` function definitely looks ugly there. I'll rework it a bit and integrate it directly into `dry_run_stats` as it seems to make more sense.